internal/controller/git/git.go

```diff
@@ -693,14 +695,17 @@ func (r *repo) setupAuth(creds RepoCredentials) error {
 	// If an SSH key was provided, use that.
 	if creds.SSHPrivateKey != "" {
 		sshConfigPath := filepath.Join(r.homeDir, ".ssh", "config")
+		err := os.MkdirAll(filepath.Dir(sshConfigPath), 0700)
+		if err != nil {
+			return fmt.Errorf("error creating SSH config directory %q: %w", sshConfigPath, err)
+		}
+		rsaKeyPath := filepath.Join(r.homeDir, ".ssh", "id_rsa")
```

Comment:
```suggestion
		sshPath := filepath.Join(r.homeDir, ".ssh")
		if err := os.Mkdir(sshPath, 0700); err != nil {
			return fmt.Errorf("error creating SSH directory %q: %w", sshPath, err)
		}
		sshConfigPath := filepath.Join(sshPath, "config")
		// nolint: lll
		const sshConfig = "Host *\n  StrictHostKeyChecking no\n  UserKnownHostsFile=/dev/null"
		if err :=
			os.WriteFile(sshConfigPath, []byte(sshConfig), 0600); err != nil {
			return fmt.Errorf("error writing SSH config to %q: %w", sshConfigPath, err)
		}

		rsaKeyPath := filepath.Join(sshPath, "id_rsa")
```

Reply:
I've incorporated some of your suggestions, but have kept the IdentityFile in the `sshConfig`.
If we're always rendering a distinct SSH config we might as well rely on its contents so that we don't have to also pass in the identity file as an option in `GIT_SSH_COMMAND`.

Thoughts?

Reply:
Yep. As mentioned in another comment, removing that piece was accidental.